src/pybind/rados/rados.pyx

```diff
@@ -1889,12 +1890,31 @@ cdef class WriteOp(object):
         with nogil:
              rados_write_op_writesame(self.write_op, _to_write, _data_len, _write_len, _offset)
 
+    def cmpext(self, cmp_buf: bytes, offset: int = 0):
+        """
+        Ensure that given object range (extent) satisfies comparison
+        :param cmp_buf: buffer containing bytes to be compared with object contents
+        :param offset: object byte offset at which to start the comparison
+        """
+
+        cdef:
+            char *_cmp_buf = cmp_buf
+            size_t _cmp_buf_len = len(cmp_buf)
+            uint64_t _offset = offset
+        with nogil:
+            rados_write_op_cmpext(self.write_op, _cmp_buf, _cmp_buf_len, _offset, &self.cmpext_result)
+     
+    def get_cmpext_result(self):
+        ret = self.cmpext_result
+        return ret 
```

Comment:
this would force the user to check `get_cmpext_result()` if `cmpext()` is included in an op. so instead of adding a dedicated method for checking it, i'm in favor of adding an exception for it as suggested by https://github.com/ceph/ceph/pull/38543#discussion_r540909524 .

Reply:
@tchaikov updated, please check